arch/arm/src/samv7/sam_usbdevhs.c

```diff
@@ -457,8 +457,8 @@ static uint32_t sam_getreg(uintptr_t regaddr);
 static void   sam_putreg(uint32_t regval, uintptr_t regaddr);
 static void   sam_dumpep(struct sam_usbdev_s *priv, int epno);
 #else
-static inline uint32_t sam_getreg(uintptr_t regaddr);
-static inline void sam_putreg(uint32_t regval, uintptr_t regaddr);
+static inline uint32_t sam_getreg(uint32_t regaddr);
+static inline void sam_putreg(uint32_t regval, uint32_t regaddr);
```

Comment:
let's keep the inline and non-inline with the same prototype, but fix the sam_getreg/sam_putreg prototype at line 856.